### PR TITLE
Track changes in cyipopt API

### LINKED ISSUE #1912: cyipopt v1.0.3 Release Issue
A new version of `cyipopt` was released today, and it is now causing both a warning message and a failure:

```
C:\Miniconda\envs\test\lib\site-packages\ipopt\__init__.py:13: FutureWarning: The module has been renamed to 'cyipopt' from 'ipopt'. 
Please import using 'import cyipopt' and remove all uses of 'import ipopt' in your code as this will be deprecated
 in a future release.

======================================================================
ERROR: test_status_maps (pyomo.contrib.pynumero.tests.test_cyipopt_examples.TestPyomoCyIpoptSolver)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/pyomo/pyomo/pyomo/contrib/pynumero/tests/test_cyipopt_examples.py", line 48, in test_status_maps
    self.assertEqual(len(cyipopt_core.STATUS_MESSAGES),
AttributeError: module 'cyipopt' has no attribute 'STATUS_MESSAGES'
```

### PR DESCRIPTION
**Pull request by jsiirola**
## Summary/Motivation:
Beginning in 1.0.3, cyipopt has changed it's API.  This PR updates our importer to work with old and new API layouts.

## Changes proposed in this PR:
- Update cyipopt importer to generate a common layout for all versions of cyipopt
- Update reference to main cyipopt GitHub page
- Update `attempt_import()`'s `DeferredImportError` to include the original import error message (to assist with tracking down dependencies)

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: pyomo/contrib/pynumero/interfaces/tests/test_dynamic_model.py [hunk 1]
```diff
@@ -368,9 +368,12 @@ def max_outflow(m, t):
 
     return m2
 
+
 class TestGreyBoxModel(unittest.TestCase):
 
-    @unittest.skipIf(not ipopt_available, "CyIpopt needed to run tests with solve")
+    @unittest.skipIf(
+        not pyo.SolverFactory('ipopt').available(exception_flag=False),
```
Comment (mrmundt) [inline pyomo/contrib/pynumero/interfaces/tests/test_dynamic_model.py:375]: Is this one actually `ipopt` or is it supposed to be `cyipopt`?

File: pyomo/contrib/pynumero/tests/test_cyipopt_examples.py [hunk 2]
```diff
@@ -37,18 +37,17 @@
...
+if not cyipopt_solver.cyipopt_available:
+    raise unittest.SkipTest(
+        "PyNumero needs CyIpopt installed to run CyIpopt tests")
 import cyipopt as cyipopt_core
 
 example_dir = os.path.join(this_file_dir(), '..', 'examples')
 
 class TestPyomoCyIpoptSolver(unittest.TestCase):
     def test_status_maps(self):
-        self.assertEqual(len(cyipopt_core.STATUS_MESSAGES),
```
Comment (mrmundt) [inline pyomo/contrib/pynumero/tests/test_cyipopt_examples.py:48]: Why remove them?

File: pyomo/contrib/pynumero/interfaces/tests/test_dynamic_model.py (hunk 1 above)
Comment (jsiirola) [reply pyomo/contrib/pynumero/interfaces/tests/test_dynamic_model.py:375]: This is really`ipopt`: the test is using the `'ipopt'` solver and not `'cyipopt'`

File: pyomo/contrib/pynumero/tests/test_cyipopt_examples.py (hunk 2 above)
Comment (jsiirola) [reply pyomo/contrib/pynumero/tests/test_cyipopt_examples.py:48]: We now have a case where we are mapping more descriptions than a specific version of cyipopt defines (we hold mappings for 1.0.3 and <= 0.3.0).  So, our map will always be bigger.  It is sufficient to catch that we can map all the statuses that this cyipopt version returns.

**Review by mrmundt — APPROVED** [review]

**Pull request merged**